### ISSUE TITLE
Unable to Move a Rectangular Selection Over Multiple Layers.

### ISSUE THREAD

**Issue by leegrey** [aseprite/aseprite#2162]
Hi, I was hoping to use the feature that was recently added to Aseprite in this commit: 
https://github.com/aseprite/aseprite/commit/3e478d3efa745789cbc054c767947bc9c2a6c3e7
The referenced feature-requests (especially #1238) seem to say that it should be possible to move a selection (created by the marquee or lasso tools) over multiple layers. However, I have found that the entire layer moves, ignoring the selection.

Please see attached screencap gif.

![Aseprite_Drag_Selection_Layers](https://user-images.githubusercontent.com/323022/65773012-905b8280-e18f-11e9-8862-e36380b89d1b.gif)

Also, please note that the Marquee tool also ignores the multiple layer selection. This mode would probably be the most ideal and fluid to use, as it is a common action when moving selections on a single layer.

![Aseprite_marquee_move_layers](https://user-images.githubusercontent.com/323022/65773983-7622a400-e191-11e9-8505-0d014225a575.gif)

### Aseprite and System version

* Aseprite version: 1.2.15 beta Steam Version (also tried 1.2.9 beta)
* System: Windows 8.1

Thanks :)

**Comment by dacap** [comment]
Hi @leegrey!

The Move tool is used to move the entire layer content (not the selection). To move only the selection you have to use a selection-like tool like in your second case.

About the movement of the selection on multiple layers, actually it works (or should work 😅), but the feedback is not immediate, you have to drop the selection to see the final result of all layers transformed. This is a known issue (first item in #2144) that we have to fix in future version, but wasn't fixed in this version because it's quite complex.

**Comment by KashouC** [comment]
Actually in the second screenshot it behaves differently than it does for me. When he goes to move the selection WITH the marquee tool after selecting both cels, the non-active layer gets deselected when he clicks. For me, it stays selected even after I press escape to "finish" the transform.

![testt](https://user-images.githubusercontent.com/17066488/65775097-cf61e780-e13f-11e9-9cee-cb616602482a.gif)

**Comment by leegrey** [comment]
As far as I can tell, it doesn't function as expected, even after letting go. The multi-layer selection is also lost...

![SelectionToolMultipleLayers](https://user-images.githubusercontent.com/323022/65775322-1083e700-e194-11e9-80d1-a0c95d1bc629.gif)

**Comment by KashouC** [comment]
Are you certain you're running 1.2.15 at the moment? Just getting that out of the way in case it's as easy a miss as that.

Otherwise, I wonder if it could be caused by old settings files or something like that?

**Comment by leegrey** [comment]
Hi @KashouC. I was on the beta version: 1.2.15beta and 1.2.9 beta.

I just switched back to the non portable version, and the behaviour is the same. Tomorrow I might try out the humble bundle version and the portable version.

But... Ah! After reading your comment again, I realise I can get the transform to work by hitting the ESC key. That's certainly a big improvement, thanks for the tip! :)  Also, thanks @dacap :) This is good enough for me, cheers!

![Escape_to_apply](https://user-images.githubusercontent.com/323022/65776044-67d68700-e195-11e9-86fa-e1d0018dc372.gif)

**Comment by dacap** [comment]
Glad to know that it's working 👍

**Comment by golddotasksquestions** [comment]
This is still quite counter-intuitive, though. As OP stated in the original post, I would also expect the move tool to move only my selection, if I have a selection. 

- If I have a selection and more than one active layer I would expect the move tool to also move my selection area on all the active layers.
- If I have a selection and more than one active layer and more than one active frame, I would expect my move tool to also move the selected areas in all active frames and in all active layers. 
- If I don't have a selection, I would expect my move tool to move everything on the active layer(s)
see https://github.com/aseprite/aseprite/issues/2002

**Comment by KashouC** [comment]
Probably right, but at the same time the marquee tool handles selection based movement and the move tool moves things more generally, so changing the move tool to work just like the marquee tool would just remove functionality. But in this case I don't really have much of an opinion either way tbh.

**Comment by golddotasksquestions** [comment]
I disagree. Good UX design allows mutiple paths to the same goal. Bad UX design allows only one valid path out of seemingly many.

**Comment by dacap** [comment]
I agree. That some users find confusing that they cannot move only the selection pixels with the move tool indicates that something is not working as expected.

We could add the possibility to start the whole selection movement cycle with the move tool. The only problem that I would need some help is to analyse if there is something we're missing. For example, I see several use cases for the move tool:

1. Move tool to move the whole layer (just like right now), e.g. if we click outside the selection with the Move tool, what should happen? should we move the layer below the mouse cursor, or the selection, or it depends on the "Auto Select Layer" option:

   <img width="275" alt="Screen Shot 2019-10-11 at 17 38 15" src="https://user-images.githubusercontent.com/39654/66683483-ee876b80-ec4d-11e9-8f74-801b68c6a68c.png">

2. Or should we move everything? E.g. The layer + the selection: it would be like right now but with the extra behavior that selection borders are moved too (no pixels are changed, only the cel and the selection border). Not sure if this would be better than the first case.
3. What about slices? should the Move tool move slices (like the slices tool already do that). What should happen if there is a layer, selection, and slices, should we move the selection or the slice if we click outside the selection and inside the slice? or slices do not make sense with the Move tool.
4. As in this discussion, if we drag-and-drop the selection with the move tool it's just like moving the selection only.

I would say that the Move tool is in its simplest form at the moment, it could be improved a lot (e.g. selecting several layers with a box like selection, transforming several layers+slices  at the same time which may be related to #2173 too, etc.). Anyway is something that must be thought with patience.

**Comment by golddotasksquestions** [comment]
My intuition is:
Whenever I can use the move tool, I should be able to use it to move whatever is the current selected layer, frame, selection subarea of a frame/layer, or slice.
Otherwise, why is it called move tool? Why can I select it but can't use it?

It does not mean that other tools like the marquee tool should not automatically and temporary change to the move tool (or imitate it's functionality) in certain cases.
For instance, I would expect if I make a selection and while I have the marquee tool as my current tool, if I move the cursor over the already selected area, I could left click and drag all pixels within that selection. In Photoshop this moves the marquee. To move the selection + content of the selection, you need to press Ctrl+LMB drag. Since Photoshop has so well known and wide spread UX, I would really not mind if Aseprite would behave the same. Especially because there seems to be no reason why Aseprite should be any different, imho.

Here is how the move tool works in Photoshop (just one layer active, just LMB pressed):
![aseprite_move_sel01](https://user-images.githubusercontent.com/47016402/66691130-71331980-ec94-11e9-8850-41851c853977.gif)

- LMB drag outside the selection will move the whole layer
- LMB drag inside the selection will move only the selected area
- Hovering over the selection will show a "scissors" move icon to indicate click-drag now will cut and move the content of the selection.

Here is how the marquee tool works:
![aseprite_marquee_sel01](https://user-images.githubusercontent.com/47016402/66691207-f0c0e880-ec94-11e9-9104-4890f504b5a9.gif)
- LMB drag outside to draw a selection area
- Hovering over the selection will show a "marquee area" move icon to indicate click-drag will now move the selected area
- LMB drag inside will therefore move the selected area
- Hovering over the selection area and pressing Ctrl will show a "sissor" move icon to indicate click-drag now will cut and move the content of the selection.
- LMB drag + Ctrl will therefore move only the selected area.

**Comment by javier-montoya** [comment]
This issue persists on v1.3.13 when dragging the Marquee tool selection with multiple layers selected. It only moves the selection for one layer at a time. The same is true for rotation. It defeats the purpose of selecting multiple layers while having an area selected since it only acts on one layer.
